src/sshfs_mount/sftp_server.cpp

```diff
@@ -242,10 +245,26 @@ void check_sshfs_status(mp::SSHSession& session, mp::SSHProcess& sshfs_process)
     }
 }
 
-auto create_sshfs_process(mp::SSHSession& session, const std::string& source, const std::string& target)
+auto create_sshfs_process(mp::SSHSession& session, mp::SshfsMount::SshfsPkgType sshfs_pkg_type,
+                          const std::string& source, const std::string& target)
 {
-    auto sshfs_process = session.exec(fmt::format(
-        "sudo sshfs -o slave -o nonempty -o transform_symlinks -o allow_other :\"{}\" \"{}\"", source, target));
+    std::string options{"-o slave -o transform_symlinks -o allow_other"};
+    std::string sshfs_exec{"sshfs"};
+
+    if (sshfs_pkg_type == mp::SshfsMount::SshfsPkgType::debian)
+    {
+        // FIXME: Need query the libfuse version to see if `nonempty` is valid.
+        // The option was removed in libfuse 3.0
+        options += " -o nonempty";
+    }
+    else
+    {
+        // TODO: Need to see if there are any new options that could be a benefit
+
+        sshfs_exec = fmt::format("LD_LIBRARY_PATH={} {}/sshfs", multipass_sshfs_lib_path, multipass_sshfs_bin_path);
```

Comment:
Could this go into a wrapper inside the snap somehow?

Reply:
Good idea, I'll look into that.

Reply:
Upon investigation, it's not trivial to use a wrapper in a snap using the `bare` base, ie, a static package.  `/bin/sh` is not available without having to do tricks.  That said, I have extracted the `LD_LIBRARY_PATH` out of the environment and just add that in the exec line.